api/src/main/java/org/pmiops/workbench/workspaces/WorkspaceServiceImpl.java

```diff
@@ -329,6 +334,28 @@ public DbWorkspace setPublished(
     return pruneInaccessibleRecentWorkspaces(userRecentWorkspaces, userId);
   }
 
+  /**
+   * Throw ForbiddenException if : a) The workspace being access if Controlled Tier and b) The
+   * logged in user doesnt have controlled Tier Access
+   *
+   * @param dbWorkspace
+   */
+  private void validateControlledTierAccess(DbWorkspace dbWorkspace) {
+    // Return if the workspace is not Controlled Tier
+    if (!dbWorkspace
+        .getCdrVersion()
+        .getAccessTier()
+        .getShortName()
+        .equals(AccessTierService.CONTROLLED_TIER_SHORT_NAME)) {
+      return;
+    }
+    // Get logged in user's access Tiers
+    List<String> accessTiers = accessTierService.getAccessTierShortNamesForUser(userProvider.get());
+    if (!accessTiers.contains(AccessTierService.CONTROLLED_TIER_SHORT_NAME)) {
```

Comment:
If we change this to:
```
String workspaceTier = dbWorkspace
        .getCdrVersion()
        .getAccessTier()
        .getShortName();
if (!accessTiers.contains(workspaceTier))
```
Then we can make this method fully generic.  Maybe name it `validateTierAccess()` then

Reply:
Thats a good idea! done